databricks/koalas/frame.py

```diff
@@ -7247,6 +7248,40 @@ def append(
 
         return cast(DataFrame, concat([self, other], ignore_index=ignore_index))
 
+    def combine_first(self, other: "DataFrame") -> "DataFrame":
+        if isinstance(other, ks.Series):
+            other = other.to_frame()
+
+        update_columns = list(
+            set(self._internal.column_labels).intersection(set(other._internal.column_labels))
+        )
+
+        update_sdf = combine_frames(self, other)._internal.resolved_copy.spark_frame
+
+        for column_labels in update_columns:
+            column_name = self._internal.spark_column_name_for(column_labels)
+            old_col = scol_for(update_sdf, "__this_" + column_name)
+            new_col = scol_for(
+                update_sdf, ("__that_" + other._internal.spark_column_name_for(column_labels))
+            )
+            update_sdf = update_sdf.withColumn(
+                "__this_" + column_name, F.when(old_col.isNull(), new_col).otherwise(old_col)
+            )
+            update_sdf = update_sdf.drop("__that_" + column_labels[0])
+
+        all_column_labels = []
+        for column in update_sdf.columns:
+            if column.startswith("__this_") or column.startswith("__that_"):
+                all_column_labels.append((column[7:],))
+
+        internal = InternalFrame(
+            spark_frame=update_sdf,
+            index_spark_column_names=list(self._internal.index_spark_column_names),
+            column_labels=list(all_column_labels),
+        )
```

Comment:
Think we can simply get the parameters for creating `InternalFrame` from `combined_df` if we created it before with a just little modifying?